lib/workers/repository/update/pr/changelog/release-notes.ts

```diff
@@ -107,9 +111,20 @@ export async function getReleaseNotes(
   if (is.undefined(matchedRelease)) {
     // no exact match of a release then check other cases
     matchedRelease = releases.find(
-      (r) => r.tag === version || r.tag === `v${version}`
+      (r) =>
+        r.tag === version ||
+        r.tag === `v${version}` ||
+        r.tag === gitRef ||
+        r.tag === `v${gitRef}`
     );
   }
+  if (is.undefined(matchedRelease) && config.extractVersion) {
+    const extractVersionRegEx = regEx(config.extractVersion);
+    matchedRelease = releases.find((r) => {
+      const extractedVersion = extractVersionRegEx.exec(r.tag)?.groups?.version;
+      return version === extractedVersion;
+    });
+  }
```

Comment:
This looks like a separate fix to me, so should be done in a separate pr

Reply:
Created https://github.com/renovatebot/renovate/pull/15962 and already reverted changes for extractVersion flow